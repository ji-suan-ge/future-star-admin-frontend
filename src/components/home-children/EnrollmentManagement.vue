<template>
  <el-container v-if="semesterState==='0'">
    <el-header>
      <el-row>
        <h3>暂无进行中学期，点击下方按钮添加新学期</h3>
      </el-row>
    </el-header>
    <el-main>
      <router-link to="/add-semester">
        <el-image :src="require('../../assets/addIcon.png')">
        </el-image>
      </router-link>
    </el-main>
  </el-container>
  <el-container v-else-if="semesterState==='1'">
    <el-header>
      <el-row type="flex" align="middle">
        <el-col :span="5">
          <h2>{{ semester }}</h2>
        </el-col>
        <el-col :span="1">
          <el-button type="primary" size="mini" class="btn" icon="el-icon-more" @click="lookOverSemesterDetail()" circle></el-button>
        </el-col>
      </el-row>
    </el-header>
    <el-main class="main-in-enrollment">
      <el-tabs v-model="activeName">
        <el-tab-pane label="未开放" name="first">
          <unopened-classes></unopened-classes>
        </el-tab-pane>
        <el-tab-pane label="招生中" name="second">
          <enrolling-classes></enrolling-classes>
        </el-tab-pane>
      </el-tabs>
    </el-main>
  </el-container>
</template>

<script>
import UnopenedClasses from '../class/ClassesUnopened'
import EnrollingClasses from '../class/ClassesEnrolling'
export default {
  name: 'EnrollmentManagementPage',
  components: {
    UnopenedClasses,
    EnrollingClasses
  },
  data () {
    return {
      activeName: 'first',
      semester: '第九期未来之星创新院',
      semesterState: '1'
    }
  },
  methods: {
    lookOverSemesterDetail: function () {
      this.$router.push('/semester-detail')
    }
  }
}
</script>

<style scoped>
  .main-in-enrollment {
    padding: 0;
    margin: 0;
  }

  .el-header {
    padding: 0;
  }
</style>
