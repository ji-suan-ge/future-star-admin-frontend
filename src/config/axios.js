import Vue from 'vue'
import axios from 'axios'
import VueAxios from 'vue-axios'
import globalConstant from '../constant/globalConstant'

Vue.use(VueAxios, axios)

axios.defaults.baseURL = globalConstant.host
axios.defaults.withCredentials = true
