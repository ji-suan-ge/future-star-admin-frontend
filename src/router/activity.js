import CreateActivityPage from '../page/CreateActivityPage'
import ShowActivityParticipantsPage from '../page/ShowActivityParticipantsPage'
import EditExistingActivityPage from '../page/EditExistingActivityPage'
import ShowActivityInformationPage from '../page/ShowActivityInformationPage'

export default [
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  },
  {
    path: '/show-participants',
    name: 'ShowActivityParticipantsPage',
    component: ShowActivityParticipantsPage
  },
  {
    path: '/edit-activity',
    name: 'EditExistingActivityPage',
    component: EditExistingActivityPage
  },
  {
    path: '/show-activity-details',
    name: 'ShowActivityInformationPage',
    component: ShowActivityInformationPage
  }
]
