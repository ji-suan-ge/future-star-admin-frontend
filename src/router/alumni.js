import ShowAlumniInformationPage from '../page/alumni-management/ShowAlumniInformationPage'
import EditAlumniInformationPage from '../page/alumni-management/EditAlumniInformationPage'
import ShowAlumniEditHistoryPage from '../page/alumni-management/ShowAlumniEditHistoryPage'
import ShowAlumniHistoryInformationPage from '../page/alumni-management/ShowAlumniHistoryInformationPage'

export default [
  {
    path: '/show-all-information',
    name: 'ShowAlumniInformationPage',
    component: ShowAlumniInformationPage
  },
  {
    path: '/edit-alumni-information',
    name: 'EditAlumniInformationPage',
    component: EditAlumniInformationPage
  },
  {
    path: '/show-edit-history',
    name: 'ShowAlumniEditHistoryPage',
    component: ShowAlumniEditHistoryPage
  },
  {
    path: '/show-history-information',
    name: 'ShowAlumniHistoryInformationPage',
    component: ShowAlumniHistoryInformationPage
  }
]
