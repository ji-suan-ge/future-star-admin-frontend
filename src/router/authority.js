import AddAdministratorPage from '../page/authority-management/AddAdministratorPage'

export default [
  {
    path: '/add-administrator',
    name: 'AddAdministratorPage',
    component: AddAdministratorPage
  }
]
