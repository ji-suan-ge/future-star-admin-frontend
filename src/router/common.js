import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import homeChildren from './home-children'

export default [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      ...homeChildren
    ]
  }
]
