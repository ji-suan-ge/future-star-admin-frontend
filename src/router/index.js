import Vue from 'vue'
import VueRouter from 'vue-router'
import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import EnrollmentManagementPage from '../page/EnrollmentManagementPage'
import CourseManagementPage from '../page/CourseManagementPage'
import ActivityManagementPage from '../page/ActivityManagementPage'
import AlumniManagementPage from '../page/AlumniManagementPage'
import AuthorityManagementPage from '../page/AuthorityManagementPage'
import CreateActivityPage from '../page/CreateActivityPage'
import AddAdministrator from '../page/AddAdministrator'
import WelcomePage from '../page/WelcomPage'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      {
        path: 'welcome',
        name: 'WelcomePage',
        component: WelcomePage
      },
      {
        path: 'enrollment-management',
        name: 'EnrollmentManagementPage',
        component: EnrollmentManagementPage
      },
      {
        path: 'course-management',
        name: 'CourseManagementPage',
        component: CourseManagementPage
      },
      {
        path: 'activity-management',
        name: 'ActivityManagementPage',
        component: ActivityManagementPage
      },
      {
        path: 'alumni-management',
        name: 'AlumniManagementPage',
        component: AlumniManagementPage
      },
      {
        path: 'authority-management',
        name: 'AuthorityManagementPage',
        component: AuthorityManagementPage
      },
      {
        path: 'add-administrator',
        name: 'AddAdministrator',
        component: AddAdministrator
      }
    ]
  },
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  }
]

const router = new VueRouter({
  routes
})

export default router
