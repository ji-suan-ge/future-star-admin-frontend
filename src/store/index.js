import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    activeIndexOfNavigation: '0',
    activeNameOfCourse: 'first',
    activeNameOfEnrollment: 'first',
    activeNameOfActivity: 'first',
    activeSemesterOfCourse: '',
    semesters: '',
    currentSemester: '',
    classes: '',
    currentClass: '',
    courses: '',
    currentCourse: '',
    infoOfAnAlumni: null
  },
  mutations: {
    changeActiveIndexOfNavigation (state, activeIndex) {
      state.activeIndexOfNavigation = activeIndex
    },
    changeActiveNameOfEnrollment (state, activeName) {
      state.activeNameOfEnrollment = activeName
    },
    changeActiveNameOfCourse (state, activeName) {
      state.activeNameOfCourse = activeName
    },
    changeActiveNameOfActivity (state, activeName) {
      state.activeNameOfActivity = activeName
    },
    changeActiveSemesterOfCourse (state, activeID) {
      state.activeSemesterOfCourse = activeID
    },
    changeSemesters (state, semesters) {
      state.semesters = semesters
    },
    changeInfoOfAnAlumni (state, alumniInfo) {
      state.infoOfAnAlumni = alumniInfo
    },
    changeCurrentSemester (state, semester) {
      state.currentSemester = semester
    },
    changeClasses (state, classes) {
      state.classes = classes
    },
    changeCurrentClass (state, currentClass) {
      state.currentClass = currentClass
    },
    changeCourses (state, courses) {
      state.courses = courses
    },
    changeCurrentCourse (state, currentCourse) {
      state.currentCourse = currentCourse
    },
    initializeStateWhileLogin (state) {
      state.activeIndexOfNavigation = '0'
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    },
    initializeStateWhileSwitch (state) {
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    }
  },
  actions: {
    changeActiveIndexOfNavigation (context, activeIndex) {
      context.commit('changeActiveIndexOfNavigation', activeIndex)
    },
    changeActiveNameOfEnrollment (context, activeName) {
      context.commit('changeActiveNameOfEnrollment', activeName)
    },
    changeActiveNameOfCourse (context, activeName) {
      context.commit('changeActiveNameOfCourse', activeName)
    },
    changeActiveNameOfActivity (context, activeName) {
      context.commit('changeActiveNameOfActivity', activeName)
    },
    changeActiveSemesterOfCourse (context, activeID) {
      context.commit('changeActiveSemesterOfCourse', activeID)
    },
    changeSemesters (context, semesters) {
      context.commit('changeSemesters', semesters)
    },
    changeInfoOfAnAlumni (context, alumniInfo) {
      context.commit('changeInfoOfAnAlumni', alumniInfo)
    },
    changeCurrentSemester (context, semester) {
      context.commit('changeCurrentSemester', semester)
    },
    changeClasses (context, classes) {
      context.commit('changeClasses', classes)
    },
    changeCurrentClass (context, currentClass) {
      context.commit('changeCurrentClass', currentClass)
    },
    changeCourses (context, courses) {
      context.commit('changeCourses', courses)
    },
    changeCurrentCourse (context, currentCourse) {
      context.commit('changeCurrentCourse', currentCourse)
    },
    initializeStateWhileLogin (context) {
      context.commit('initializeStateWhileLogin')
    },
    initializeStateWhileSwitch (context) {
      context.commit('initializeStateWhileSwitch')
    }
  },
  getters: {
    getActiveIndexOfNavigation (state) {
      return state.activeIndexOfNavigation
    },
    getActiveNameOfEnrollment (state) {
      return state.activeNameOfEnrollment
    },
    getActiveNameOfCourse (state) {
      return state.activeNameOfCourse
    },
    getActiveNameOfActivity (state) {
      return state.activeNameOfActivity
    },
    getActiveSemesterOfCourse (state) {
      return state.activeSemesterOfCourse
    },
    getSemesters (state) {
      return state.semesters
    },
    getInfoOfAnAlumni (state) {
      return state.infoOfAnAlumni
    },
    getCurrentSemester (state) {
      return state.currentSemester
    },
    getClasses (state) {
      return state.classes
    },
    getCurrentClass (state) {
      return state.currentClass
    },
    getCourses (state) {
      return state.courses
    },
    getCurrentCourse (state) {
      return state.currentCourse
    }
  },
  modules: {
  }
})
