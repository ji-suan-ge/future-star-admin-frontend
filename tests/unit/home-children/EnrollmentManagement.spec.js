import { createLocalVue, mount } from '@vue/test-utils'
import EnrollmentManagement from '../../../src/components/home-children/EnrollmentManagement'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import VueRouter from 'vue-router'
import axios from 'axios'
import VueAxios from 'vue-axios'
import store from '../../../src/store'
import router from '../../../src/router'

describe('EnrollmentManagement', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(ElementUI)
  localVue.use(VueRouter)
  localVue.use(VueAxios, axios)
  const wrapper = mount(EnrollmentManagement, {
    store,
    router,
    localVue
  })
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
