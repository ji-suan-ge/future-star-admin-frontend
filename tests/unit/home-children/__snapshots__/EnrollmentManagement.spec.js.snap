// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrollmentManagement 快照测试 1`] = `
<section class="el-container is-vertical">
  <header class="el-header" style="height: 60px;">
    <div class="el-row is-align-middle el-row--flex">
      <div class="el-col el-col-5">
        <h2>第九期未来之星创新院</h2>
      </div>
      <div class="el-col el-col-1"><button type="button" class="el-button btn el-button--primary el-button--mini is-circle">
          <!----><i class="el-icon-more"></i>
          <!----></button></div>
    </div>
  </header>
  <main class="el-main main-in-enrollment">
    <div class="el-tabs el-tabs--top el-tabs--border-card">
      <div class="el-tabs__header is-top">
        <div class="el-tabs__nav-wrap is-top">
          <div class="el-tabs__nav-scroll">
            <div role="tablist" class="el-tabs__nav is-top" style="transform: translateX(-0px);">
              <div id="tab-first" aria-controls="pane-first" role="tab" aria-selected="true" tabindex="0" class="el-tabs__item is-top is-active">未开放</div>
              <div id="tab-second" aria-controls="pane-second" role="tab" tabindex="-1" class="el-tabs__item is-top">招生中</div>
            </div>
          </div>
        </div>
      </div>
      <div class="el-tabs__content">
        <div role="tabpanel" id="pane-first" aria-labelledby="tab-first" class="el-tab-pane">
          <div>
            <!---->
            <div class="el-col el-col-12">
              <div class="el-card is-always-shadow">
                <div class="el-card__header">
                  <div class="el-row is-align-middle el-row--flex">
                    <div class="el-col el-col-16">
                      素质教育专题班
                    </div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-check"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-more"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-edit-outline"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
                        <!----><i class="el-icon-delete"></i>
                        <!----></button></div>
                  </div>
                </div>
                <div class="el-card__body">
                  <div class="el-row">
                    起止日期：2019/12/20—— 2019/12/21
                  </div>
                  <div class="el-row">
                    招生人数：0 / 15
                  </div>
                  <div class="el-row">
                    班级状态：未开放
                  </div>
                  <div class="el-row">
                    班级简介：这里是一个加了长文本省略号替代的班级介绍
                  </div>
                </div>
              </div>
            </div>
            <div class="el-col el-col-12">
              <div class="el-card is-always-shadow">
                <div class="el-card__header">
                  <div class="el-row is-align-middle el-row--flex">
                    <div class="el-col el-col-16">
                      素质教育专题班
                    </div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-check"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-more"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-edit-outline"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
                        <!----><i class="el-icon-delete"></i>
                        <!----></button></div>
                  </div>
                </div>
                <div class="el-card__body">
                  <div class="el-row">
                    起止日期：2019/12/20—— 2019/12/21
                  </div>
                  <div class="el-row">
                    招生人数：0 / 15
                  </div>
                  <div class="el-row">
                    班级状态：未开放
                  </div>
                  <div class="el-row">
                    班级简介：这里是一个加了长文本省略号替代的班级介绍
                  </div>
                </div>
              </div>
            </div>
            <div class="el-col el-col-12">
              <div class="el-card is-always-shadow">
                <div class="el-card__header">
                  <div class="el-row is-align-middle el-row--flex">
                    <div class="el-col el-col-16">
                      素质教育专题班
                    </div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-check"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-more"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-edit-outline"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
                        <!----><i class="el-icon-delete"></i>
                        <!----></button></div>
                  </div>
                </div>
                <div class="el-card__body">
                  <div class="el-row">
                    起止日期：2019/12/20—— 2019/12/21
                  </div>
                  <div class="el-row">
                    招生人数：0 / 15
                  </div>
                  <div class="el-row">
                    班级状态：未开放
                  </div>
                  <div class="el-row">
                    班级简介：这里是一个加了长文本省略号替代的班级介绍
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div role="tabpanel" aria-hidden="true" id="pane-second" aria-labelledby="tab-second" class="el-tab-pane" style="display: none;">
          <div>
            <!---->
            <div class="el-col el-col-12">
              <div class="el-card is-always-shadow">
                <div class="el-card__header">
                  <div class="el-row is-align-middle el-row--flex">
                    <div class="el-col el-col-14">
                      素质教育专题班
                    </div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-check"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-more"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-edit-outline"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-s-custom"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
                        <!----><i class="el-icon-delete"></i>
                        <!----></button></div>
                  </div>
                </div>
                <div class="el-card__body">
                  <div class="el-row">
                    起止日期：2019/12/20—— 2019/12/21
                  </div>
                  <div class="el-row">
                    招生人数：0 / 15
                  </div>
                  <div class="el-row">
                    班级状态：招生中
                  </div>
                  <div class="el-row">
                    班级简介：这里是一个加了长文本省略号替代的班级介绍
                  </div>
                </div>
              </div>
            </div>
            <div class="el-col el-col-12">
              <div class="el-card is-always-shadow">
                <div class="el-card__header">
                  <div class="el-row is-align-middle el-row--flex">
                    <div class="el-col el-col-14">
                      素质教育专题班
                    </div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-check"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-more"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-edit-outline"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-s-custom"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
                        <!----><i class="el-icon-delete"></i>
                        <!----></button></div>
                  </div>
                </div>
                <div class="el-card__body">
                  <div class="el-row">
                    起止日期：2019/12/20—— 2019/12/21
                  </div>
                  <div class="el-row">
                    招生人数：0 / 15
                  </div>
                  <div class="el-row">
                    班级状态：招生中
                  </div>
                  <div class="el-row">
                    班级简介：这里是一个加了长文本省略号替代的班级介绍
                  </div>
                </div>
              </div>
            </div>
            <div class="el-col el-col-12">
              <div class="el-card is-always-shadow">
                <div class="el-card__header">
                  <div class="el-row is-align-middle el-row--flex">
                    <div class="el-col el-col-14">
                      素质教育专题班
                    </div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-check"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-more"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-edit-outline"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--primary el-button--small is-circle">
                        <!----><i class="el-icon-s-custom"></i>
                        <!----></button></div>
                    <div class="el-col el-col-2"><button type="button" class="el-button el-button--danger el-button--small is-circle">
                        <!----><i class="el-icon-delete"></i>
                        <!----></button></div>
                  </div>
                </div>
                <div class="el-card__body">
                  <div class="el-row">
                    起止日期：2019/12/20—— 2019/12/21
                  </div>
                  <div class="el-row">
                    招生人数：0 / 15
                  </div>
                  <div class="el-row">
                    班级状态：招生中
                  </div>
                  <div class="el-row">
                    班级简介：这里是一个加了长文本省略号替代的班级介绍
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</section>
`;
